src/app/portlets/content-types/fields/content-type-fields-variables/content-type-fields-variables.component.html

```diff
@@ -9,8 +9,8 @@
     <p-column field="value" [header]="messages['contenttypes.field.variables.value_header.label']" [editable]="true"></p-column>
     <p-column styleClass="content-type-fields__variables-actions" [header]="messages['contenttypes.field.variables.actions_header.label']" >
         <ng-template let-fieldVar="rowData" let-index="rowIndex" pTemplate="body">
-            <dot-icon-button icon="save" (click)="saveVariable(fieldVar, index)" [attr.disabled]="canSaveField(index) ? true : null"></dot-icon-button>
-            <dot-icon-button icon="delete" (click)="deleteVariable(index)"></dot-icon-button>
+            <dot-icon-button icon="delete_outline" (click)="deleteVariable(index)"></dot-icon-button>
+            <dot-icon-button icon="edit" (click)="saveVariable(fieldVar, index)" [attr.disabled]="canSaveField(index) ? true : null"></dot-icon-button>
```

Comment:
```canSaveField(index)``` should't this method return `true` or `false`? also is bad to add logic to the templates, check 16: https://medium.freecodecamp.org/best-practices-for-a-clean-and-performant-angular-application-288e7b39eb6f

Reply:
@fmontes  that method indeed return true or false. 

But this is a _special case_ `[attr.disabled]="canSaveField(index) ? true : null"` and on *false* we need to return `null` otherwise it will render disabled="false" witch will be interprete it as `true`

Reply:
to remove it from the template, I guess this can be changed:

```canSaveField(index: number): any {
        return (this.fieldVariables[index].key === '' || this.fieldVariables[index].value === '') ? true : null;
    }```

but now it will return `any` and not a typed value

Reply:
Can you just set the value to a variable and pass the variable to the attr?

Reply:
not sure how, without overcomplicating the code

Reply:
this is what I ended up with:
```
canSaveField(index: number): boolean | null {
        return this.fieldVariables[index].key === '' || this.fieldVariables[index].value === '';	        return (this.fieldVariables[index].key === '' || this.fieldVariables[index].value === '') ? true : null;
    }	    }
```